plugins/modules/host.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+)
+
+from cm_client import ClustersResourceApi
+from cm_client import HostsResourceApi
+from cm_client.rest import ApiException
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: host
+short_description: Manage hosts within Cloudera Manager
+description:
+  - Create, delete, attach or detach a host instance.
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+"""
+DOCUMENTATION = r"""
+---
+module: host
+short_description: Manage hosts within Cloudera Manager
+description:
+  - Allows for the management of hosts within the Cloudera Manager.
+  - It provides functionalities to create, delete, attach, or detach host instance from a cluster.
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+"""
+
+EXAMPLES = r"""
+---
+- name: Create a host
+  cloudera.cluster.host:
+    host: "example.cloudera.host"
+    username: "will_jordan"
+    password: "S&peR4Ec*re"
+    host_name: "Ecs_node_01"
+    host_ip: "10.9.8.7"
+    state: "present"
+- name: Attach a host to the Cluster
```

Comment:
Separate the examples with new lines for readability

Reply:
Is this an example of attaching an existing host to a cluster?

Reply:
No this is just creating a host as the state is `present` , for attaching to a cluster there is a state `attach`